Mage.Sets/src/mage/cards/s/SparkshaperVisionary.java

```diff
@@ -0,0 +1,99 @@
+package mage.cards.s;
+
+import mage.MageInt;
+import mage.abilities.TriggeredAbility;
+import mage.abilities.common.BeginningOfCombatTriggeredAbility;
+import mage.abilities.common.DealsCombatDamageToAPlayerTriggeredAbility;
+import mage.abilities.effects.common.continuous.BecomesCreatureTargetEffect;
+import mage.abilities.effects.keyword.ScryEffect;
+import mage.abilities.keyword.FlyingAbility;
+import mage.abilities.keyword.HexproofAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Duration;
+import mage.constants.SubType;
+import mage.constants.TargetController;
+import mage.filter.common.FilterControlledPermanent;
+import mage.filter.common.FilterControlledPlaneswalkerPermanent;
+import mage.game.permanent.token.TokenImpl;
+import mage.target.common.TargetControlledPermanent;
+
+import java.util.UUID;
+
+/**
+ * @author Susucr
+ */
+public final class SparkshaperVisionary extends CardImpl {
+
+    public static final FilterControlledPermanent filter =
+        new FilterControlledPlaneswalkerPermanent("planeswalkers you control");
+
+    public SparkshaperVisionary(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{2}{U}");
+
+        this.subtype.add(SubType.HUMAN);
+        this.subtype.add(SubType.WIZARD);
+        this.power = new MageInt(0);
+        this.toughness = new MageInt(5);
+
+        // At the beginning of combat on your turn, choose any number of target planeswalkers you control. Until end of turn, they become 3/3 blue Bird creatures with flying, hexproof, and "Whenever this creature deals combat damage to a player, scry 1."
+        TriggeredAbility triggeredAbility =
+            new BeginningOfCombatTriggeredAbility(
+                new BecomesCreatureTargetEffect(
+                    new SparkshaperVisionaryToken(),
+                    false, false, Duration.EndOfTurn,
+                    false, false, true
```

Comment:
I believe this should be keeping abilities. Might want to add the reminder text as well.

Reply:
Oh I did miss that `keepAbilities` intent. `losaAllAbilities` to true was keeping the abilities in my test.

For the reminder text, sure I'll add it.

Reply:
I'm not totally sure I'm following the constructors correctly, they're kind of messy. If you've tested and it works properly that's fine.

Reply:
This is the final values, I suspect the only difference would be for gaining types from static ability? really not sure.
![becometarget](https://github.com/magefree/mage/assets/34709007/76f3a66f-28ec-4212-8971-e92d0bf7ec21)

And it indeed works as far as I can tell. I had many Jaces living the Bird life (I did remove the scry reminder in the meantime)
 
![jacebird](https://github.com/magefree/mage/assets/34709007/952941e3-7fca-4917-a8e5-37160fe068fe)